source/auth/tests/legacy/connection-string.yml

```diff
@@ -350,59 +350,30 @@ tests:
     mechanism: MONGODB-AWS
     mechanism_properties:
       AWS_SESSION_TOKEN: token!@#$%^&*()_+
-- description: should recognise the mechanism and request callback (MONGODB-OIDC)
-  uri: mongodb://localhost/?authMechanism=MONGODB-OIDC
-  callback:
-  - oidcRequest
+- description: should recognise the mechanism with aws provider (MONGODB-OIDC)
+  uri: mongodb://localhost/?authMechanism=MONGODB-OIDC&authMechanismProperties=PROVIDER_NAME:aws
   valid: true
   credential:
     username:
     password:
     source: "$external"
     mechanism: MONGODB-OIDC
     mechanism_properties:
-      REQUEST_TOKEN_CALLBACK: true
-- description: should recognise the mechanism when auth source is explicitly specified
-    and with request callback (MONGODB-OIDC)
-  uri: mongodb://localhost/?authMechanism=MONGODB-OIDC&authSource=$external
-  callback:
-  - oidcRequest
+      PROVIDER_NAME: aws
+- description: should recognise the mechanism when auth source is explicitly specified and with provider (MONGODB-OIDC)
+  uri: mongodb://localhost/?authMechanism=MONGODB-OIDC&authSource=$external&authMechanismProperties=PROVIDER_NAME:aws
   valid: true
   credential:
     username:
     password:
     source: "$external"
     mechanism: MONGODB-OIDC
     mechanism_properties:
-      REQUEST_TOKEN_CALLBACK: true
-- description: should recognise the mechanism with request and refresh callback (MONGODB-OIDC)
-  uri: mongodb://localhost/?authMechanism=MONGODB-OIDC
+      PROVIDER_NAME: aws
+- description: should ignore username and password if specified for aws provider (MONGODB-OIDC)
+  uri: mongodb://user:pass@localhost/?authMechanism=MONGODB-OIDC&authMechanismProperties=PROVIDER_NAME:aws
```

Comment:
PROVIDER_NAME and callbacks are mutual exclusive. Callback parameter probably has to be removed from this test.

Reply:
You're correct. The `callback` legacy test case parameter was never documented and is not supported by most drivers, so I intended to remove it from all test cases here, but accidentally left that one in. I've updated that test case to assert that providing a password causes a validation error, which matches the spec.